lib/dependency-manager-adapters/npm.js

```diff
@@ -82,26 +93,35 @@ module.exports = CoreObject.extend({
       return null;
     }
   },
-  _install() {
+  _install(depSet) {
     let adapter = this;
     let mgrOptions = this.managerOptions || [];
+    let cmd = this.useYarnCommand ? 'yarn' : 'npm';
 
-    debug('Run npm install with options %s', mgrOptions);
+    // buildManagerOptions overrides all default
+    if (typeof this.buildManagerOptions === 'function') {
+      mgrOptions = this.buildManagerOptions(depSet);
 
-    let cmd = this.useYarnCommand ? 'yarn' : 'npm';
-    if (this.useYarnCommand) {
-      if (mgrOptions.indexOf('--no-lockfile') === -1) {
-        mgrOptions = mgrOptions.concat(['--no-lockfile']);
+      if (!Array.isArray(mgrOptions)) {
+        throw new Error('buildManagerOptions must return an array of options');
       }
-      // npm warns on incompatible engines
-      // yarn errors, not a good experience
-      if (mgrOptions.indexOf('--ignore-engines') === -1) {
-        mgrOptions = mgrOptions.concat(['--ignore-engines']);
+    } else {
+      if (this.useYarnCommand) {
```

Comment:
Do we intend to mandate the buildManagerOptions in the future? If so, we can add a deprecation here that encourages to use the manager options for these defaults.

Reply:
> Do we intend to mandate the buildManagerOptions in the future?

No, I believe that it will always be a "power user" option. I would expect that **most** users would use whatever our default command line options are, and you'd only specify `buildManagerOptions` if you had to deviate from that for some reason.

Reply:
I agree; it is backward compatible and users do not need to use it if there isn't a need.